### PR TITLE
fixed - Search by resource ID with space fails with 500

### PR DESCRIPTION
**Pull request by shabd67**
# Resolves

[SALUS-1073](https://jira.rax.io/browse/SALUS-1073)

# What

Search by resource ID with space fails with 500

### CODE REVIEW

File: public/src/main/java/com/rackspace/salus/telemetry/api/web/ResourcesController.java [hunk 1]
```diff
@@ -71,12 +73,11 @@ public ResourcesController(ServicesProperties servicesProperties) {
...
       @RequestHeader HttpHeaders headers,
-      @PathVariable String resourceId) {
-
-    final String backendUri = UriComponentsBuilder
+      @PathVariable String resourceId) throws UnsupportedEncodingException {
+    String backendUri = UriComponentsBuilder
         .fromUriString(servicesProperties.getResourceManagementUrl())
         .path("/api/tenant/{tenantId}/resources/{resourceId}")
-        .buildAndExpand(tenantId, resourceId)
+        .buildAndExpand(tenantId, URLEncoder.encode(resourceId,"UTF-8"))
```
Comment (itzg) [inline public/src/main/java/com/rackspace/salus/telemetry/api/web/ResourcesController.java:80]: We used to have some issues with double URL encoding. In fact, I think it was one of your early tasks to use the `toUriString` throughout to consistently address those. Can you work with @harageth to confirm this explicit encoding is necessary and why it wasn't encoding properly as it was?

**Comment by itzg** [comment]
Also, please don't remove the "How" section from the pull request template. That is probably the most important section to write in PRs. Case in point, I am now left wondering what you even found and why you're manually doing URL encode/decode that Spring MVC should be doing.

**Comment by harageth** [comment]
@shabd67 did you make any progress looking into the `UriComponentsBuilder` to see what it had to say about encodings? The value should be encoded already as it arrives in the public API. Have you debugged that to see what the value is when it gets to the public API?

**Review by itzg — APPROVED** [review]
Much cleaner. Thanks for digging in further.

**Pull request merged**